**Rework pass — consuming the failed first dispatch (run 31968806742).**

## Diagnosis (the `validate` FAILURE is a credential, not a code path)
The dispatch ran the workflow end-to-end and failed loud exactly as designed, but on auth:

```
19:51:26Z  CVM deployment failed.
19:51:26Z  Message: Invalid API key
19:51:26Z  Request: POST https://cloud-api.phala.com/api/v1/cvms/provision
19:51:26Z  HTTP: 401 Unauthorized
19:51:27Z .. 20:02:38Z  [phase1 1..60/60] cvm status: absent / err
20:02:38Z  ##[error]CVM did not reach running (last=err)
```

- Secret plumbing is **correct**: the workflow reads `secrets.PHALA_API_KEY` (which exists, set 2026-05-28) into the CLI's `PHALA_CLOUD_API_KEY` env — no name mismatch; the `***` in the log is the real secret, masked.
- Two independent transports reject the same credential: the GitHub runner (repo secret) and this box's only phala profile (`amiller-users-projects` — `phala status`/`whoami` → `Invalid API key`). The **key value itself is stale/revoked**.
- No CVM was created (billing impact: zero), and the run proved live in CI: dispatch trigger, CLI install, secret resolution, the 10-min poll ceiling, the fail-loud error, and teardown/orphan-sweep with nothing to sweep.

## Patch pushed (`2908f46`)
The dispatch burned its whole 10-min poll on an instant, definitive 401. Phase 1 now exits in ~1s with an actionable `::error::` when the deploy log matches `HTTP: 40[13]|Invalid API key|Unauthorized`; every other failure mode keeps the #13 slow-Phala poll untouched. Tier 0 (CI-only): YAML parse + `bash -n` on all 7 inline scripts re-run green, grep pattern tested against the verbatim failed-run log (match) and a benign provisioning log (no match). Evidence: `.evidence/issue-14/06-dispatch-31968806742.txt`, `07-auth-fastfail.txt`, EVIDENCE.md addendum.

## BLOCKED — need from operator:
A fresh Phala Cloud API key, in either order:
1. Mint a new key (Phala Cloud console → API keys, or `phala login` on the box) — I cannot create one; it requires your Phala account identity.
2. Update the repo secret: `gh secret set PHALA_API_KEY --repo teleport-computer/shape-rotator-matrix` (and, if you want box-side parity, refresh `~/.phala-cloud/credentials.json` / re-`phala login`).
3. Re-dispatch **ephemeral staging validation** on this branch once. With a valid key it should now either go green (create → running → smoke → teardown ≈ 5–10 min) or fail fast with a real, actionable message.

Until that dispatch runs green, this PR stays honestly unstaged for merge (no `ready-to-merge` added) — the live round-trip is acceptance item 2 and it has not happened yet. Billing-semantics confirmation (acceptance item 1) is likewise an operator dashboard check.

--- a/.evidence/issue-14/08-fastfail-live.txt
+++ b/.evidence/issue-14/08-fastfail-live.txt
@@ -41,3 +41,11 @@ conclusion: FAILURE — intended outcome: definitive 401 on a stale credential
    (gh secret list); this box's only phala profile is likewise rejected (phala status →
    Invalid API key). The key is revoked Phala-side — no credential held on this box can
    perform the round-trip. Minting a new key is operator-only (Phala account identity).
+
+## Re-confirmation — run 31971800063 (pull_request, d468197, 2026-08-16T20:52:13Z)
+Deterministic re-run ~40 min later on the same stale secret: 27 s total, identical fast-fail
+(20:52:35.05Z ##[error]phala rejected our credentials...), "no orphaned staging CVMs",
+always() teardown fired, no CVM created, zero spend. Note: the pull_request paths filter is
+matched against the FULL PR diff (three-dot vs base), so every push to this PR re-triggers
+validation regardless of which files changed — intended per acceptance #2 ("on every
+deploy-relevant PR"), and free while the credential is stale (fails fast, nothing provisioned).

--- a/.evidence/issue-14/08-fastfail-live.txt
+++ b/.evidence/issue-14/08-fastfail-live.txt
@@ -1,0 +1,43 @@
+# Run 31969904168 — live confirmation of the auth fast-fail (patch 2908f46)
+repo:      teleport-computer/shape-rotator-matrix
+workflow:  staging-validate.yml ("ephemeral staging validation")
+event:     pull_request (auto-triggered by the push of 2908f46 itself — the fast-fail patch)
+head:      ready-14 @ 2908f46
+created:   2026-08-16T20:13:05Z
+completed: 2026-08-16T20:13:29Z  (~24 s total)
+conclusion: FAILURE — intended outcome: definitive 401 on a stale credential
+
+## Pre-flight orphan sweep (runs before phase 1)
+20:13:24.86Z  no orphaned staging CVMs
+              → run 31968806742 left nothing stranded; the guaranteed-teardown design held.
+
+## Phase 1 — create ephemeral CVM, poll until running
+20:13:24.94Z  $ phala deploy --api-token *** --name staging-pr-31969904168 \
+                  -c docker-compose.staging.yml -e .staging.env > deploy1.log
+20:13:26.12Z  Provisioning CVM staging-pr-31969904168...
+20:13:26.12Z  CVM deployment failed.
+20:13:26.12Z  Message: Invalid API key
+20:13:26.12Z  Request: POST https://cloud-api.phala.com/api/v1/cvms/provision
+20:13:26.12Z  HTTP: 401 Unauthorized
+20:13:26.13Z  ##[error]phala rejected our credentials (401/403) — the PHALA_API_KEY secret is
+              stale or invalid. Polling cannot fix this; refresh the secret and re-dispatch.
+20:13:26.13Z  ##[error]Process completed with exit code 1.
+              → fast-fail fired 1.2 s into phase 1. Pre-patch behavior (run 31968806742) burned
+                the full 60×10 s = 10 min poll on the same instant 401.
+
+## Always-teardown (+ orphan sweep), `if: always()`
+20:13:26.15Z  ##[notice]tearing down staging-pr-31969904168
+20:13:27.74Z  Delete CVM failed. Message: Invalid API key
+20:13:27.74Z  Request: GET https://cloud-api.phala.com/api/v1/cvms/staging-pr-31969904168 → 401
+              → expected: no CVM was ever created, nothing exists to delete. The step still ran,
+                proving the always() teardown hook fires on the fast-fail exit path.
+
+## Facts established by this run
+1. The 2908f46 auth fast-fail is verified LIVE in CI — previously it was only grep-tested
+   against the captured 31968806742 log; now observed firing on a real runner.
+2. The pre-flight orphan sweep and the always() teardown both execute on the fast-fail path.
+3. No CVM was provisioned → zero Phala spend for this run.
+4. Credential state at run time: repo secret PHALA_API_KEY last updated 2026-05-28T15:02:46Z
+   (gh secret list); this box's only phala profile is likewise rejected (phala status →
+   Invalid API key). The key is revoked Phala-side — no credential held on this box can
+   perform the round-trip. Minting a new key is operator-only (Phala account identity).

--- a/.evidence/issue-14/EVIDENCE.md
+++ b/.evidence/issue-14/EVIDENCE.md
@@ -60,3 +60,11 @@ watched dispatch, called out explicitly in the PR:
 First dispatch (run 31968806742) failed: stale PHALA_API_KEY -> 401 at provision (06). Patched
 phase 1 to fail fast on definitive auth rejection; re-ran checks 01/02 equivalents on the patched
 file (07). Blocked on operator: fresh Phala API key in the PHALA_API_KEY secret, then one re-dispatch.
+
+--- 2026-08-16 addendum 2 (rework pass 2 on PR #69) ---
+The patch's own push triggered run 31969904168 (pull_request event): the fast-fail fired live,
+1.2 s into phase 1, with the actionable error; orphan sweep + always() teardown both ran; no CVM
+created, zero spend (08). Fast-fail is now CI-observed, not just grep-tested. Credential still
+revoked Phala-side (repo secret unchanged since 2026-05-28; box profile rejected too) — every
+acceptance path (#2 round-trip, #3 broken-compose run) requires a key only the operator can mint.
+Still blocked; no ready-to-merge set.

